### DISCUSSION TITLE
Windows Update broke AdGuard Home

### DISCUSSION THREAD

**Discussion by Pradhansb** [AdguardTeam/AdGuardHome#3309] [Q&A]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * 
* **How did you install AdGuard Home:**
  * 
* **How did you setup DNS configuration:**
  * 
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:**
  * 
* **Operating system and version:**
  * 

### Screenshots

<details><summary>Screenshot:</summary>

</details>

**Comment by Pradhansb** [comment]
Hello! I am not sure this is the correct place to post this. About 3 weeks ago, I installed AdGuard Home on a Windows machine. I changed the DNS in the router and the ipV4 in Windows. Everything was working perfectly. Then Windows updated, and I totally lost the internet. I ended up reinstalling AdGuardHome from scratch. I notice that under the properties of each network, you can now change the DNS servers for IPv4. SOOO...now my hard-wired cxn no longer works. My wireless cxns seem to work fine. 
I want to make sure I am doing the right thing. I am not clear as to whether changing the IPv4 which is listed under the Windows setup is essential? Or is that only for AdGuard Windows.
Secondly, I have tried to change the IPv4 address on my hard-wired cxn both on the adapter and in the new Property box. Nothing works. As soon as I do this, I lose the internet on that machine.

Like I said, all was fine just 2 days ago! Forgive me if this is not clear/or if I am wordy.
Thanks,
Pradhan

**Comment by ainar-g** [comment]
Hello.  What do you mean when you say that you've changed the DNS on your router?  If you plan to only use AGH on that Windows machine, you shouldn't change any router settings.

**Reply by Pradhansb** [reply]
Sorry, I replied to your email with a detailed response thinking it would post. Apparently not.  So, I followed the instructions on the Wiki page precisely, and as mentioned it WAS working fine. Allow me to be more specific with the screen capture instructions.
I installed AdGuard Home on an old notebook with freshly installed Win10. My intention was to have this as a server protecting my entire home network.
After installing, AdGuard informed me it was listening on: 192. etc...
It also then says to change the ipV4 dns to the same listening address on any Windows machine.
I have added the instructions as screen captures to this message.
As mentioned in the original email, all was working fine. I could see that ads were being intercepted everywhere.

To protect the entire network, the instructions then say to change the router DNS from auto to the listening ip, which I did

After the Windows updated (apparently), I lost the internet entirely. I could only get it back by changing the ipV4 on my Windows machines back to Auto. But at this point, AdGuard's filtering stopped.


![Snag_114890a8](https://user-images.githubusercontent.com/43834702/124441709-922d7f00-dd41-11eb-9758-3c0ff1e42013.png)
![Snag_1148d36e](https://user-images.githubusercontent.com/43834702/124441736-9b1e5080-dd41-11eb-87da-b5bde03b70d4.png)
![Snag_1148f84c](https://user-images.githubusercontent.com/43834702/124441753-a07b9b00-dd41-11eb-9206-ec04b564ba19.png)
![Snag_1149175d](https://user-images.githubusercontent.com/43834702/124441767-a5404f00-dd41-11eb-81ca-cdbce699cb4a.png)

**Comment by ainar-g** [comment]
Apologies for the long radio silence.  Is this issue still occurring on the latest betas?

**Reply by renatosampaio81** [reply]
@ainar-g
I'm having the same problem.
The other devices see the DNS ip (by ping), but I can't access the settings panel and neither the DNS service works nor other devices on the network.

I haven't tried docker, it's my next step.

**Comment by delphir-com** [comment]
@ainar-g I think I have the same (similar?) issue too, but I have a slightly different setup and a bit more details on this issue.

So in my case `AdGuard Home` is installed on a separate network device (Ubuntu server 20.04 LTS) inside the docker container (I will refer to this device as `ADGUARD-Server` in my message).
The described "no internet" Win10 issue happens because of the AdGuard DNS server "stops" responding and Windows 10 can't resolve any dns addresses.

Manually changing DNS server in Win10 Network Adapter settings to `8.8.8.8` fixes the internet, but ofcourse this way AdGuard is not taken into effect.

Other connections from Win10 machine to the `ADGUARD-Server` stops working too (in my case I was testing SSH, http and dns connection).

Whatever connections to `ADGUARD-Server` from other machines in the local network are working totally fine (was testing ssh, http and adguard dns).

Connections to other machines in the local network from the affected Win10 machine are working fine too (was testing SSH connection to some other machine in local net).

So it looks like the issue is happening just between Win10 vs `ADGUARD-Server` connections. And from `ADGUARD-Server` side I don't see any attempts for the connection.

I also noticed that this usually happens when windows is trying to do some huge amount of windows updates (e.g. after the fresh install). Then the issue might disappear "by itself" after 30-40 minutes.

I tried to reboot `ADGUARD-Server` and `Router` - this do not help.
Rebooting Win10 machine do not help too.

Previously I noticed that issue was "disappearing" if I switch Wifi network type to "Private" network. but last time this didn't help, so probably it was just a luck.

Also my "Rate limit" in AdGuard Home is set to `0`, so this should not be the case.

Any idea what may cause this?

**Reply by Pradhansb** [reply]
One thing that may be happening--and it's a simple thing, easy to overlook. It has happened a number of times with me that when Windows updates, the dns address on which AdGuard listens also updates to something different...requiring me to go into the modem to reassign the IP to match. So it might be worth checking your Setup tab and double-check that your listening IP matches the IP set in your modem.

**Reply by delphir-com** [reply]
not my case. Like I wrote - AdGuard DNS is installed on a SEPARATE machine, but windows looses an access to it.
Replacing dns on windows machine to whatever public dns IP fixes the internet, but changing dns IP back to AdGuard breaks the dns resolution functionality again.

at the same time other computers in the same local network can use this AdGuard DNS with no problems

**Reply by ainar-g** [reply]
@delphir-com, apologies for the late response.  I'm not at all sure what could cause this, but the two things worth inspecting are the Windows firewall (if one is enabled), the antivirus software (if there are any), and the [event log][1].  The latter could contain network or DNS errors, as well as some other related system alerts.

[1]: https://www.freecodecamp.org/news/event-viewer-how-to-access-the-windows-10-activity-log/
